Don't render signup link if registration feature is disabled

mgomes opened the pull request:
Hides the signup link since the controller just redirects back to root path if the registration feature is disabled.

mgomes commented:
Going to fix the system tests.

krschacht commented:
@mgomes but the system test shouldn't fail by making this change because the feature is enabled by default so the link **should** be there. I actually think your conditional is backwards :)

mgomes commented:
@krschacht Ok, fixed the logic and added a system test. I had to move some of the private methods in the feature tests to helpers. I think they'll be useful, though, as more features get introduced and need to be tested in isolation.

krschacht commented:
Good call, thanks for cleaning that up! Merging in
